test/runLint

```diff
@@ -61,3 +64,9 @@ gitgrep "$cmdstart"'unset [^-]' 'Explicitly specify "unset -v/-f"'
 
 gitgrep "$cmdstart"'((set|shopt)\s+[+-][a-z]+\s+posix\b|(local\s+)?POSIXLY_CORRECT\b)' \
     'fiddling with posix mode breaks keybindings with some bash versions'
+
+gitgrep '\$\{([^{}\n]|\{.*\})+/([^{}\n]|\{.*\})+/([^{}"\n]|\{.*\})*\$.*\}' \
+    '$rep of ${var/pat/$rep} needs to be double-quoted for shopt -s patsub_replacement (bash >= 5.2)'
```

Comment:
The commit message of 5b959a7e55fc743b59b55249d8529113294ea022 details a case why we should be rather using `_comp_compgen` than double quoting. Should we update this to suggest that, too?

Looks good to me otherwise, pre-approved and good to merge with this addressed (or not addressed if that's the right thing to do).

Reply:
This becomes long, so in commit c35af29e96dc62eb85b0306397ddee72c5441623, I added a new subsection in `doc/styleguide.md` under the quoting section and added pointers to the related lint items.